implementations/micrometer-registry-kairos/src/main/java/io/micrometer/kairos/KairosMeterRegistry.java

```diff
@@ -0,0 +1,269 @@
+                    if (m instanceof FunctionTimer) {
+                        return writeFunctionTimer((FunctionTimer) m, wallTime);
+                    }
+                    if (m instanceof FunctionCounter) {
+                        return writeFunctionCounter((FunctionCounter) m, wallTime);
+                    }
+                    if (m instanceof DistributionSummary) {
+                        return writeSummary((DistributionSummary) m, wallTime);
+                    }
+                    return writeCustomMetric(m, wallTime);
+                }).collect(Collectors.joining(", "));
+
+            sendToKairos(String.format("[%s]", payload));
+        }
+    }
+
+    private void sendToKairos(String bulkPayload) {
+        HttpURLConnection con = null;
+
+        try {
+            URL templateUrl = new URL(config.host());
+            con = (HttpURLConnection) templateUrl.openConnection();
+            con.setConnectTimeout((int) config.connectTimeout().toMillis());
+            con.setReadTimeout((int) config.readTimeout().toMillis());
+            con.setRequestMethod(HttpMethod.POST);
+            con.setRequestProperty(HttpHeader.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+            con.setDoOutput(true);
+
+            if (StringUtils.isNotBlank(config.userName()) && StringUtils.isNotBlank(config.password())) {
+                byte[] authBinary = (config.userName() + ":" + config.password()).getBytes(StandardCharsets.UTF_8);
+                String authEncoded = Base64.getEncoder().encodeToString(authBinary);
+                con.setRequestProperty(HttpHeader.AUTHORIZATION, "Basic " + authEncoded);
+            }
+            logger.trace("Sending payload to KairosDB:");
+            logger.trace(bulkPayload);
+
+            try (OutputStream os = con.getOutputStream()) {
+                os.write(bulkPayload.getBytes());
+                os.flush();
+            }
+
+            int status = con.getResponseCode();
+
+            if (status >= 200 && status < 300) {
+                logger.trace("successfully sent events to KairosDB");
+            } else {
+                logger.error("failed to send metrics, status: {} message: {}", status, con.getResponseMessage());
+            }
+        } catch (Throwable e) {
+            logger.warn("failed to send metrics", e);
+        } finally {
+            quietlyCloseUrlConnection(con);
+        }
+    }
+
+    private void quietlyCloseUrlConnection(@Nullable HttpURLConnection con) {
+        try {
+            if (con != null) {
+                con.disconnect();
+            }
+        } catch (Exception ignore) {
+        }
+    }
+
+    private static class KairosMetricBuilder {
+        private StringBuilder sb = new StringBuilder("{");
+
+        KairosMetricBuilder field(String key, String value) {
+            if (sb.length() > 1) {
+                sb.append(",");
+            }
+            sb.append("\"").append(key).append("\":\"").append(value).append("\"");
```

Comment:
append('\"') can be used to append single characters

Reply:
thanks, fixed